### DISCUSSION TITLE
extra spacing due to ScreenReaderOnly in Buttons

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#1851] [General] [answered]
Hello team, had a very minor question about using `<ScreenReaderOnly />` in a `<Button />`. In the documentation we see it being used like:

```tsx
<Button variant="secondary" size="small">
   Buy <ScreenReaderOnly>+1 (903) 603-9469</ScreenReaderOnly>
</Button>
```

This however creates a small invisible space, it looks like children within a Button get some margin between each other (see screenshots below):

![image](https://user-images.githubusercontent.com/6936736/132761343-80176fc9-98c9-4b60-8d17-dba7a69249bb.png)

![image](https://user-images.githubusercontent.com/6936736/132761963-db652a99-4940-431e-8a73-bc07bba0086e.png)

We've gotten around this issue by using this code instead:
```tsx
<Button variant="secondary" size="small">
  <Text as="span">
    Buy
    <ScreenReaderOnly> +1 (903) 603-9469</ScreenReaderOnly>
  </Text>
</Button>
```

Do we need to make any revisions to the documentation or the margin in Button children to revise this?

https://paste.twilio.design/components/screen-reader-only#providing-clear-actions-while-still-reducing-visual-repetition

**Comment by nkrantz** [comment, marked as answer]
Hi @vnguyen94, thanks for bringing this to our attention! We actually have a ticket for this that was sitting deep in the backlog (😬), we're going to talk about prioritizing it at our next backlog refinement. Keep track here: https://issues.corp.twilio.com/browse/DSYS-2188

**Reply by vankhoawin** [reply]
perfect thank you!

**Reply by nkrantz** [reply]
Hey @vnguyen94! Just here with an update that this fix was [merged in](https://github.com/twilio-labs/paste/pull/1977) today. Let us know if you have any more issues with it!
